lib/agent.js

```diff
@@ -257,3 +257,52 @@ Agent.prototype.handleUncaughtExceptions = function (cb) {
 Agent.prototype.flush = function (cb) {
   this._instrumentation.flush(cb)
 }
+
+Agent.prototype.lambda = function wrapLambda (name, fn) {
+  var agent = this
+  return function wrappedLambda (payload, context, callback) {
+    var trans = agent.startTransaction()
+    if (!trans) {
+      return fn(payload, context, callback)
+    }
+
+    trans.type = 'lambda'
+    trans.name = name
+
+    function wrapLambdaCallback (callback) {
+      return function wrappedLambdaCallback (err, result) {
+        var context = this
+
+        trans.setCustomContext({
```

Comment:
We need a better place to store this than "custom" - but that's another discussion. For now it's probably the best location

Reply:
Yeah. In this case though, the input and output are just arbitrary data, so it's not really something we can have a solid spec for other than just the presence of the keys. It's not necessarily even JSON, it could be string content too.